pkg/kdump/kdump-client.es6

```diff
@@ -0,0 +1,183 @@
+ */
+
+import cockpit from 'cockpit';
+var _ = cockpit.gettext;
+import {proxy as serviceProxy} from 'service';
+import {ConfigFile} from './config-client.es6';
+
+const deprecatedKeys = ["net", "options", "link_delay", "disk_timeout", "debug_mem_level", "blacklist"];
+const knownKeys = [
+    "raw", "nfs", "ssh", "sshkey", "path", "core_collector", "kdump_post", "kdump_pre", "extra_bins", "extra_modules",
+    "default", "force_rebuild", "override_resettable", "dracut_args", "fence_kdump_args", "fence_kdump_nodes"
+];
+
+/*  initializes the kdump status
+ *  emits "kdumpStatusChanged" when the status changes, along with a status object:
+ *  {
+ *      installed: true/false,
+ *      enabled:   true/false,
+ *      state:     current state,
+ *      config:    settings from kdump.conf
+ *      target:    dump target info, content depends on dump type
+ *                 always contains the keys:
+ *                     target          value in ["local", "nfs", "ssh", "raw", "mount", "unknown"]
+ *                     multipleTargets true if the config file has more than one target defined, false otherwise
+ *  }
+ *
+ */
+export class KdumpClient {
+    constructor() {
+        var self = this;
+        self.state = {
+            installed: undefined,
+            enabled: undefined,
+            state: undefined,
+            config: undefined,
+            target: undefined,
+        };
+        cockpit.event_target(self);
+
+        // listen to service status changes
+        self.kdumpService = serviceProxy("kdump");
+        self.kdumpService.addEventListener('changed', function() {
+            self.state.installed = self.kdumpService.exists;
+            self.state.enabled = self.kdumpService.enabled;
+            self.state.state = self.kdumpService.state;
+            self.dispatchEvent("kdumpStatusChanged", self.state);
+        });
+
+        // watch the config file
+        self.configClient = new ConfigFile("/etc/kdump.conf", true);
+        // catch config changes
+        self.configClient.addEventListener('kdumpConfigChanged', function() {
+            self.state.config = self.configClient.settings;
+            self.state.target = self.targetFromSettings(self.configClient.settings);
+            self.dispatchEvent("kdumpStatusChanged", self.state);
+        });
+    }
+    ensureOn() {
+        // we consider the state to be "on" when it's enabled and running
+        return cockpit.all(
+            this.kdumpService.enable(),
+            this.kdumpService.start()
+        );
+    }
+    ensureOff() {
+        // we consider the state to be "off" when it's disabled and stopped
+        return cockpit.all(
+            this.kdumpService.stop(),
+            this.kdumpService.disable()
+        );
+    }
+    crashKernel() {
+        // crash the system kernel
+        return cockpit.spawn(["/bin/sh", "-c", "echo 1 > /proc/sys/kernel/sysrq && echo c > /proc/sysrq-trigger"], { superuser: "require" });
```

Comment:
At a very minimum ... This need to sync disks the same way you do it in the tests.